crates/nu-parser/src/flatten.rs

```diff
@@ -264,10 +264,6 @@ pub fn flatten_expression(
         Expr::Float(_) => {
             vec![(expr.span, FlatShape::Float)]
         }
-        Expr::MatchPattern(pattern) => {
-            // FIXME: do nicer flattening later
-            flatten_pattern(pattern)
-        }
```

Comment:
The only concern I had was if removing this would degrade the syntax highlighting in `match` but running the stable version showed the same improvement worthy showing (and it is also in fact handled by `Expr::MatchBlock` if I gather the code below correctly)